isis/tests/DistanceTests.cpp

```diff
@@ -0,0 +1,295 @@
+#include "Distance.h"
+#include "Displacement.h"
+#include "IException.h" 
+#include "SpecialPixel.h"
+
+#include <gtest/gtest.h>
+
+using namespace Isis;
+
+TEST(DistanceTests, DefaultConstructor) {
+	Distance dist;
+	EXPECT_EQ(dist.meters(), Null);
+	EXPECT_EQ(dist.kilometers(), Null);
+	EXPECT_EQ(dist.pixels(), Null);
+	EXPECT_EQ(dist.solarRadii(), Null);
+}
+
+
+TEST(DistanceTests, MetersConstructor) {
+	Distance dist(1500500, Distance::Meters);
+	EXPECT_EQ(dist.meters(), 1500500);
+	EXPECT_FLOAT_EQ(dist.kilometers(), 1500.5);
+	EXPECT_FLOAT_EQ(dist.solarRadii(), 0.002155922);
+	EXPECT_EQ(dist.pixels(1), 1500500);
+}
+
+
+TEST(DistanceTests, KilometersConstructor) {
+	Distance dist(1500.5, Distance::Kilometers);
+	EXPECT_FLOAT_EQ(dist.kilometers(), 1500.5);
+	EXPECT_EQ(dist.meters(), 1500500);
+	EXPECT_FLOAT_EQ(dist.solarRadii(), 0.002155922);
+	EXPECT_EQ(dist.pixels(1), 1500500);
+}
+
+
+TEST(DistanceTests, SolarRadiiConstructor) {
+	Distance dist(0.002155922, Distance::SolarRadii);
+	EXPECT_FLOAT_EQ(dist.solarRadii(), 0.002155922);
+	EXPECT_FLOAT_EQ(dist.meters(), 1.5005e+06);
+	EXPECT_FLOAT_EQ(dist.kilometers(), 1500.5);
+	EXPECT_FLOAT_EQ(dist.pixels(1), 1.5005e+06);
+}
+
+
+TEST(DistanceTests, PixelsConstructor) {
+	Distance dist(1500500, Distance::Pixels);
+	EXPECT_EQ(dist.pixels(1), 1500500);
+	EXPECT_EQ(dist.meters(), 1500500);
+	EXPECT_FLOAT_EQ(dist.kilometers(), 1500.5);
+	EXPECT_FLOAT_EQ(dist.solarRadii(), 0.002155922);
+}
```

Comment:
See my other comment and use 1 solar radii = 6.9599e8 meters = 6.9599e5 km = 6.9599e8 pixels (with 1m per pixel) because all of the math will work out exactly.